pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -382,13 +378,12 @@ func (s *splitAndCacheMiddleware) fetchCacheExtents(ctx context.Context, keys []
 
 // storeCacheExtents stores the extents for given key in the cache.
 func (s *splitAndCacheMiddleware) storeCacheExtents(ctx context.Context, key string, tenantIDs []string, extents []Extent) {
-	ttl := resultsCacheTTL
+	ttl := validation.SmallestPositiveNonZeroDurationPerTenant(tenantIDs, s.limits.ResultsCacheTTL)
 	lowerTTLWithinTimePeriod := validation.MaxDurationPerTenant(tenantIDs, func(tenantID string) time.Duration {
 		return time.Duration(s.limits.OutOfOrderTimeWindow(tenantID))
```

Comment:
Doesn't have to be done now but `Limits.OutOfOrderTimeWindow()` should really return a `time.Duration` for consistency with all the other methods for that struct.

Reply:
Agree, will send separate PR for that.

Reply:
Created https://github.com/grafana/mimir/pull/4420.